### PR TITLE
feat(replays): Replay Player DOM Alert 

### LINKED ISSUE #40519: Replay Player area needs a label or something reminding people that they can open the inspector and see the html/css inside the replay
### Problem
Its impossible to know that you can inspect the DOM of this session

### Solution
Add a dismissible alert to educate the user on this hidden functionality. Once it's dismissed by the user, they shouldn't see alert again.

_**See latest designs in thread below...**_

---

~Text `Use your browsers developer tools to see how the DOM changes.`~

Image

Image

### PR DESCRIPTION
**Pull request by danecando**
Adds a small alert overlay over the replay player to let the user know that they can inspect the DOM of the application in the player. Alert is dismissible and the state is persisted to local storage.

Closes #40519 

## Changes
- Adds PlayerDOMAlert component
- Adds PlayerDOMAlert to ReplayPlayer

### Latest
Screen Shot 2022-11-09 at 9 24 28 AM
Screen Shot 2022-11-09 at 9 24 41 AM

### Original

Screen Shot 2022-11-08 at 11 53 46 AM
Screen Shot 2022-11-08 at 11 56 00 AM

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: static/app/components/replays/playerDOMAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  font-size: ${p => p.theme.fontSizeMedium};
+`;
+
+const DOMAlert = styled('div')`
+  display: inline-flex;
+  align-items: center;
+  justify-items: center;
+  padding: ${space(1)} ${space(2)};
+  margin: 0 ${space(1)};
+  color: ${p => p.theme.white};
```
[the remark this answers is not in the document]
Comment (danecando) [reply static/app/components/replays/playerDOMAlert.tsx:62]: I still get white on dark mode

Screen Shot 2022-11-08 at 12 33 54 PM

I did however just notice that the hover color on the dismiss button needs to be updated

File: static/app/components/replays/playerDOMAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  if (isDismissed) {
+    return null;
+  }
+
+  return (
+    <DOMAlertContainer data-test-id="player-dom-alert">
+      <DOMAlert>
+        <IconInfo size="xs" />
+        Right click and inspect your app's DOM
```
Comment (billyvg) [inline static/app/components/replays/playerDOMAlert.tsx:32]: Needs `t()`

File: static/app/components/replays/playerDOMAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        <DismissButton
+          priority="link"
+          size="sm"
+          icon={<IconClose size="xs" />}
+          aria-label="Close Alert"
```
Comment (billyvg) [inline static/app/components/replays/playerDOMAlert.tsx:37]: `t()` here too

File: static/app/components/replays/playerDOMAlert.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+export default PlayerDOMAlert;
+
+const DOMAlertContainer = styled('div')`
+  position: absolute;
+  bottom: ${space(1)};
+  left: 0;
+  width: 100%;
+  text-align: center;
+  font-size: ${p => p.theme.fontSizeMedium};
+`;
```
Comment (ryan953) [inline static/app/components/replays/playerDOMAlert.tsx:55]: can we disable pointer events on this one. There's lots of invisible space taken up by it, and if people are going to right-click then shouldn't land on our html

File: static/app/components/replays/playerDOMAlert.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  if (isDismissed) {
+    return null;
+  }
+
+  return (
+    <DOMAlertContainer data-test-id="player-dom-alert">
+      <DOMAlert>
+        <IconInfo size="xs" />
+        {t("Right click and inspect your app's DOM")}
```
Comment (ryan953) [inline static/app/components/replays/playerDOMAlert.tsx:33]:
```suggestion
        {t("Right click & inspect your app's DOM")}
```

**Review by ryan953 — APPROVED** [review]

File: static/app/components/replays/playerDOMAlert.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  text-align: center;
+  font-size: ${p => p.theme.fontSizeMedium};
+  pointer-events: none;
+`;
+
+const DOMAlert = styled('div')`
+  display: inline-flex;
+  align-items: center;
+  justify-items: center;
+  padding: ${space(1)} ${space(2)};
```
[the remark this answers is not in the document]
Comment (danecando) [reply static/app/components/replays/playerDOMAlert.tsx:62]: Removed line-height and update screenshots

**Pull request merged**